cpp/modmesh/buffer/pymod/wrap_SimpleArrayPlex.cpp

```diff
@@ -300,16 +292,64 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArrayPlex : public WrapBase<Wr
                        { 
                         const auto shape = make_shape(py_shape);
                         return array.reshape(shape); }); })
-            .def_property_readonly("has_ghost", [](wrapped_type & self)
-                                   { return execute_callback_with_typed_array(
-                                         self, [](auto & array)
-                                         { return array.has_ghost(); }); })
-            .def_property_readonly("nbody", [](wrapped_type & self)
-                                   { return execute_callback_with_typed_array(
-                                         self, [](auto & array)
-                                         { return array.nbody(); }); })
-            /// TODO: should have the same interface as WrapSimpleArray
+            .def_property_readonly("has_ghost", DECL_MM_EXECUTE_TYPED_ARRAY_METHOD(has_ghost))
+            .def_property(
+                "nghost",
+                // getter
+                DECL_MM_EXECUTE_TYPED_ARRAY_METHOD(nghost),
+                // setter
+                [](wrapped_type & self, size_t size)
+                { return execute_callback_with_typed_array(
+                      self, [size](auto & array)
+                      { return array.set_nghost(size); }); })
+            .def_property_readonly("nbody", DECL_MM_EXECUTE_TYPED_ARRAY_METHOD(nbody))
+            .wrap_modifiers()
+            .wrap_calculators()
+            // ATTENTION: always keep the same interface between WrapSimpleArrayPlex and WrapSimpleArray
+            ;
+    }
+
+    wrapper_type & wrap_modifiers()
+    {
+        namespace py = pybind11; // NOLINT(misc-unused-alias-decls)
+
+        (*this)
+            .def(
+                "fill",
+                [](wrapped_type & self, pybind11::object const & py_value)
+                {
+                    auto datatype = self.data_type();
+                    execute_callback_with_typed_array(
+                        self, [&py_value, datatype](auto & array)
+                        {
+                            using value_type = typename std::remove_reference_t<decltype(array[0])>;
+                            verify_python_value_datatype(py_value, datatype);
+                            const auto value = py_value.cast<value_type>();
+                            array.fill(value);
+                        }); },
+                py::arg("value"))
+            //
+            ;
+
+        return *this;
+    }
+
+    wrapper_type & wrap_calculators()
+    {
+#define DECL_MM_EXECUTE_TYPED_ARRAY_METHOD_RETUN_TYPED_VALUE(typed_array_method) \
+    [](wrapped_type & self) { return execute_callback_with_typed_array(          \
+                                  self, [](auto & array) { return pybind11::cast(array.typed_array_method()); }); }
```

Comment:
the key is using `pybind11::cast` for the return type to avoid type differences.

Reply:
Could you please double check if clang-format is doing a right thing here?  If it is, leave it here.  The format does not look nice, while tolerable.

Reply:
@yungyuc  clang-format seems not working well for all the cases in which we put the lambda in the pybind11 class definition.